src/codec/MapPutCodec.ts

```diff
@@ -0,0 +1,59 @@
+import {ClientMessage} from '../ClientMessage';
+import {BitsUtil} from '../BitsUtil';
+import {Utils} from './Utils';
+import {CustomCodec} from './CustomCodec';
+import {MapMessageType} from './MapMessageType';
+
+var REQUEST_TYPE = MapMessageType.MAP_PUT;
+var RESPONSE_TYPE = 105;
+var RETRYABLE = false;
+
+
+export class MapPutCodec{
+
+constructor() {
+}
+
+
+
+
+static calculateSize(name : string , key : Buffer , value : Buffer , threadId : any , ttl : any ){
+    // Calculates the request payload size
+    var dataSize : number = 0;
+    dataSize += Utils.calculateSizeString(name);
+    dataSize += Utils.calculateSizeBuffer(key);
+    dataSize += Utils.calculateSizeBuffer(value);
+    dataSize += BitsUtil.LONG_SIZE_IN_BYTES;
+    dataSize += BitsUtil.LONG_SIZE_IN_BYTES;
+    return dataSize;
+}
+
+static encodeRequest(name : string, key : Buffer, value : Buffer, threadId : any, ttl : any){
+    // Encode request into clientMessage
+    var payloadSize : number;
+    var clientMessage = ClientMessage.newClientMessage(payloadSize=this.calculateSize(name, key, value, threadId, ttl));
+    clientMessage.setMessageType(REQUEST_TYPE);
+    clientMessage.setRetryable(RETRYABLE);
+    clientMessage.appendString(name);
+    clientMessage.appendBuffer(key);
+    clientMessage.appendBuffer(value);
+    clientMessage.appendLong(threadId);
+    clientMessage.appendLong(ttl);
+    clientMessage.updateFrameLength();
+    return clientMessage;
+}
+
+static decodeResponse(clientMessage : ClientMessage){
+    // Decode response from client message
+    var parameters :any ;
```

Comment:
Should initialize this like `parameters: any = {}`. Otherwise `parameters['response'] = ...` fails.